src/dashboard/src/components/archival_storage/tests/test_archival_storage.py

```diff
@@ -200,6 +204,154 @@ def test_get_pointer_known_pointer(
+
+    Test the new route via the Archival Storage tab to be able to
+    download the Elasticsearch AIP index as a CSV. Here we make sure
+    that various headers are set as well as testing whether or not the
+    data is returned correctly.
+    """
+    dashboard_login_and_setup(client, django_user_model, username, password)
+    CSV_MIME = "text/csv"
+    RESULT_FILENAME = "test-filename.csv"
+    RESULT_DISPOSITION = 'attachment; filename="{}"'.format(RESULT_FILENAME)
+    ordered_headers = [
+        "Name",
+        "UUID",
+        "AICID",
+        "Count AIPs in AIC",
+        "Size",
+        "File count",
+        "Accession IDs",
+        "Created date (UTC)",
+        "Status",
+        "Type",
+        "Encrypted",
+        "Location",
+    ]
+    mock_augmented_result = [
+        {
+            "status": "Stored",
+            "encrypted": False,
+            "AICID": "AIC#2040",
+            "countAIPsinAIC": 2,
+            "accessionids": [],
+            "uuid": "a341dbc0-9715-4806-8477-fb407b105a5e",
+            "name": "tz",
+            "created": 1594938100,
+            "file_count": 2,
+            "location": "/var/archivematica/AIPStore",
+            "isPartOf": None,
+            "size": "200.1\xa0KB",
+            "type": "AIC",
+        },
+        {
+            "status": "Stored",
+            "encrypted": True,
+            "AICID": None,
+            "countAIPsinAIC": None,
+            "accessionids": ["Àà", "Éé", "Îî", "Ôô", "Ùù"],
+            "uuid": "22423d5c-f992-4979-9390-1cb61c87da14",
+            "name": "tz",
+            "created": 1594938200,
+            "file_count": 2,
+            "location": "thé cloud",
+            "isPartOf": None,
+            "size": "152.1\xa0KB",
+            "type": "AIP",
+        },
+    ]
+    mocker.patch("elasticSearchFunctions.get_client")
+    mocker.patch("elasticsearch.Elasticsearch.search")
+    mocker.patch(
+        "components.archival_storage.views.search_augment_aip_results",
+        return_value=mock_augmented_result,
+    )
+    REQUEST_PARAMS = {
+        "requestFile": True,
+        "mimeType": CSV_MIME,
+        "fileName": RESULT_FILENAME,
+        "returnAll": True,
+    }
+    response = client.get(
+        "/archival-storage/search/?{}".format(urlencode(REQUEST_PARAMS))
+    )
+
+    # Check that our response headers are going to be useful to the caller.
+    assert response.get("content-type") == CSV_MIME
+    assert response.get("content-disposition") == RESULT_DISPOSITION
+
+    streamed_content = b"".join([content for content in response.streaming_content])
+    csv_file = StringIO(streamed_content)
+    data_frame = pd.read_csv(csv_file, header=0, encoding="utf8")
+
+    # Make sure that our headers come out as expected and in the right order.
+    assert list(data_frame.columns.values) == ordered_headers
+
+    # Make some assertions about the quality of the rest of our data,
+    # particularly those we're encoding as
```

Comment:
This comment could use a bit of editing.

Reply:
True!